libs/shared/type-site/src/lib/i-editor-context.ts

```diff
@@ -90,6 +91,8 @@ export interface IResizeData {
 }
 
 export interface IEditorContext {
+  // Reference to store used to save/restore the site, populated on initialization
+  store?: ISiteStore
```

Comment:
I spent a while trying to make all the dependencies work, but in the end I think it's worth including a reference to the store in the editor context. It's really easy to set up, and it means we can avoid importing the source/store in many places. I hope it makes sense.